src/Exceptionless.DateTimeExtensions/FormatParsers/FormatParsers/PartParsers/DateMathPartParser.cs

```diff
@@ -0,0 +1,229 @@
+
+            if (anchor.Equals("now", StringComparison.OrdinalIgnoreCase))
+            {
+                baseTime = relativeBaseTime;
+            }
+            else
+            {
+                // Parse explicit date from the date group
+                string dateStr = match.Groups["date"].Value;
+                if (!TryParseExplicitDate(dateStr, relativeBaseTime.Offset, out baseTime))
+                    return null;
+            }
+
+            // Parse and apply operations
+            string operations = match.Groups["operations"].Value;
+            var result = ApplyOperations(baseTime, operations, isUpperLimit);
+
+            return result;
+        }
+        catch
+        {
+            // Return null for any parsing errors to maintain robustness
+            return null;
+        }
+    }
+
+    /// <summary>
+    /// Attempts to parse an explicit date string with proper timezone handling.
+    /// Supports various Elasticsearch-compatible date formats with optional timezone information.
+    ///
+    /// Timezone Behavior:
+    /// - If timezone is specified (Z, +HH:MM, -HH:MM): Preserved from input
+    /// - If no timezone specified: Uses the provided fallback offset
+    ///
+    /// This matches Elasticsearch's behavior where explicit timezone information takes precedence.
+    /// </summary>
+    /// <param name="dateStr">The date string to parse</param>
+    /// <param name="offset">Fallback timezone offset for dates without explicit timezone</param>
+    /// <param name="result">The parsed DateTimeOffset with correct timezone</param>
+    /// <returns>True if parsing succeeded, false otherwise</returns>
+    private static bool TryParseExplicitDate(string dateStr, TimeSpan offset, out DateTimeOffset result)
+    {
+        result = default;
+
+        if (string.IsNullOrEmpty(dateStr))
+            return false;
+
+        // Try various formats that Elasticsearch supports
+        string[] formats = {
+            "yyyy-MM-dd",
+            "yyyy-MM-ddTHH:mm:ss",
+            "yyyy-MM-ddTHH:mm",
+            "yyyy-MM-ddTHH",
+            "yyyy-MM-ddTHH:mm:ssZ",
+            "yyyy-MM-ddTHH:mm:ss.fff",
+            "yyyy-MM-ddTHH:mm:ss.fffZ",
+            "yyyy-MM-ddTHH:mm:sszzz",
+            "yyyy-MM-ddTHH:mm:ss.fffzzz",
+            "yyyy.MM.dd",
+            "yyyy.MM.ddTHH:mm:ss",
+            "yyyy.MM.ddTHH:mm",
+            "yyyy.MM.ddTHH",
+            "yyyy.MM.ddTHH:mm:ssZ",
+            "yyyy.MM.ddTHH:mm:ss.fff",
+            "yyyy.MM.ddTHH:mm:ss.fffZ",
+            "yyyy.MM.ddTHH:mm:sszzz",
+            "yyyy.MM.ddTHH:mm:ss.fffzzz",
+            "yyyyMMdd",
+            "yyyyMMddTHHmmss",
+            "yyyyMMddTHHmm",
+            "yyyyMMddTHH",
+            "yyyyMMddTHHmmssZ",
+            "yyyyMMddTHHmmss.fff",
+            "yyyyMMddTHHmmss.fffZ",
+            "yyyyMMddTHHmmsszzz",
+            "yyyyMMddTHHmmss.fffzzz"
+        };
+
+        foreach (string format in formats)
```

Comment:
this could be very expensive.

Reply:
Yes it could and it's silly. It really should just support ISO format and it surely does not need to support milliseconds.